app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -458,6 +466,12 @@ public Study updateStudy(Study study, boolean isAdminUpdate) {
 
         Validate.entityThrowingException(validator, study);
 
+        // Only admins can delete or modify upload metadata fields. Check this after validation, so we don't have to
+        // deal with duplicates.
+        if (!isAdminUpdate) {
+            checkUploadMetadataConstraints(originalStudy, study);
+        }
```

Comment:
But client devs can add?

Reply:
Yes. Added to the comment "Anyone (admin or developer) can add or re-order fields."